kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationFilter.java

```diff
@@ -138,10 +140,12 @@ public CompletionStage<RequestFilterResult> onSaslAuthenticateRequest(short apiV
                                     .setErrorMessage(e.getMessage())
                                     .setAuthBytes(request.authBytes());
                             LOGGER.debug("SASL Authentication failed : {}", e.getMessage(), e);
+                            context.clientSaslAuthenticationFailure(OAUTHBEARER_MECHANISM, null, (Exception) e.getCause());
```

Comment:
Nicer to do `if (e.getCause() instanceof SaslAuthenticationException cause) {` a few lines above?